python/lsst/ip/isr/isrTaskLSST.py

```diff
@@ -0,0 +1,1219 @@
+
+            # Overscan correction on amp-by-amp basis.
+            if amp.getRawHorizontalOverscanBBox().isEmpty():
+                self.log.info("ISR_OSCAN: No overscan region.  Not performing overscan correction.")
+                overscans.append(None)
+            else:
+
+                # Perform overscan correction on subregions.
+                overscanResults = self.overscan.run(ccdExposure, amp)
+
+                self.log.debug("Corrected overscan for amplifier %s.", amp.getName())
+                if len(overscans) == 0:
+                    ccdExposure.getMetadata().set('OVERSCAN', "Overscan corrected")
+
+                overscans.append(overscanResults if overscanResults is not None else None)
+
+        return overscans
+
+    def snapCombine(self, **kwargs):
+        # TODO DM 36638
+        pass
+
+    def getLinearizer(self, detector):
+        # Here we assume linearizer as dict or LUT are not supported
+        # TODO DM 28741
+
+        # TODO construct isrcalib input
+        linearizer = linearize.Linearizer(detector=detector, log=self.log)
+        self.log.warning("Constructing linearizer from cameraGeom information.")
+
+        return linearizer
+
+    def gainNormalize(self, **kwargs):
+        # TODO DM 36639
+        gains = []
+        readNoise = []
+
+        return gains, readNoise
+
+    def updateVariance(self, ampExposure, amp, overscanImage=None, ptcDataset=None):
+        """Set the variance plane using the gain and read noise
+
+        The read noise is calculated from the ``overscanImage`` if the
+        ``doEmpiricalReadNoise`` option is set in the configuration; otherwise
+        the value from the amplifier data is used.
+
+        Parameters
+        ----------
+        ampExposure : `lsst.afw.image.Exposure`
+            Exposure to process.
+        amp : `lsst.afw.cameraGeom.Amplifier` or `FakeAmp`
+            Amplifier detector data.
+        overscanImage : `lsst.afw.image.MaskedImage`, optional.
+            Image of overscan, required only for empirical read noise.
+        ptcDataset : `lsst.ip.isr.PhotonTransferCurveDataset`, optional
+            PTC dataset containing the gains and read noise.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if either ``usePtcGains`` or ``usePtcReadNoise``
+            are ``True``, but ptcDataset is not provided.
+
+            Raised if ```doEmpiricalReadNoise`` is ``True`` but
+            ``overscanImage`` is ``None``.
+
+        See also
+        --------
+        lsst.ip.isr.isrFunctions.updateVariance
+        """
+        maskPlanes = [self.config.saturatedMaskName, self.config.suspectMaskName]
+        if self.config.usePtcGains:
+            if ptcDataset is None:
+                raise RuntimeError("No ptcDataset provided to use PTC gains.")
+            else:
+                gain = ptcDataset.gain[amp.getName()]
+                self.log.info("Using gain from Photon Transfer Curve.")
+        else:
```

Comment:
This should maybe be a warning that you are using a questionable gain value.

Reply:
Sorry, the warning should be below for `amp.getGain()` and there should _not_ be a warning for the usual gain from the PTC.